app/models/pipeline_run.rb

```diff
@@ -493,8 +493,9 @@ def monitor_results
 
     # Get pipeline_version, which determines S3 locations of output files.
     # If pipeline version is not present, we cannot load results yet.
+    # Except, if the pipeline run is finalized, we have to (this is a failure case).
     update_pipeline_version(self, :pipeline_version, pipeline_version_file)
-    return if pipeline_version.blank?
+    return if pipeline_version.blank? && !finalized
```

Comment:
@jameshowardwang  I am sneaking a tricker one past you.   This was actually responsible for some of the problems in the seed env, and can also happen in production if the batch job gets killed before it starts, or otherwise fails to produce a version file.  The result monitor would keep iterating over that job, without ever marking its status as "failed".   With this fix, it will check the (lack of) results, and mark it as failed.

Reply:
😮

Reply:
very nice catch, thanks boris!